### PR TITLE
🌱  Replace CAPI Matcher with Controller Runtime implementation

### LINKED ISSUE #6841: Use matchers from Controller Runtime komega
Today we use a matching library kept in `internal/test/matchers`. This work has now been upstreamed (thanks to @schrej :slightly_smiling_face: ) in controller runtime `pkg/envtest/komega`

We should get rid of our matchers and replace their usage with the komega matchers as soon as it's available in a controller runtime release that we're using in CAPI.

/area testing

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Remove the Cluster API EqualObject matcher implementation and replace with the Controller Runtime implementation added in Controller Runtime.
Fixes #6841

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
I don't think there's any need to add this to e.g. release notes as the change is internal.

File: internal/controllers/topology/cluster/reconcile_state_test.go [hunk 1]
```diff
@@ -46,12 +47,11 @@ import (
 	"sigs.k8s.io/cluster-api/internal/hooks"
 	fakeruntimeclient "sigs.k8s.io/cluster-api/internal/runtime/client/fake"
 	"sigs.k8s.io/cluster-api/internal/test/builder"
-	. "sigs.k8s.io/cluster-api/internal/test/matchers"
 )
 
 var (
 	IgnoreNameGenerated = IgnorePaths{
-		{"metadata", "name"},
+		"metadata.name",
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/reconcile_state_test.go:54]: Q: Should we use ObjectMeta.Name here instead to consistently use the go type format?

**Review by sbueringer — COMMENTED** [review]
Nice!

File: internal/controllers/topology/cluster/reconcile_state_test.go (hunk 1 above)
Comment (killianmuldoon) [reply internal/controllers/topology/cluster/reconcile_state_test.go:54]: The Go style format doesn't work with unstructured objects - even for metadata name as far as I can see.

Comment (sbueringer) [reply internal/controllers/topology/cluster/reconcile_state_test.go:54]: Ah alright. Does the "JSON"  format work in the other cases?

Comment (killianmuldoon) [reply internal/controllers/topology/cluster/reconcile_state_test.go:54]: It does - I've updated to make all the IgnorePaths in json format.

File: test/extension/handlers/topologymutation/handler_test.go [hunk 2]
```diff
@@ -304,7 +304,7 @@ func TestHandler_GeneratePatches(t *testing.T) {
 
 			// Expect all response fields to be as expected. responseItems are ignored here and tested below.
 			// Ignore the message to not compare error strings.
-			g.Expect(tt.expectedResponse).To(EqualObject(response, IgnorePaths{{"items"}, {"message"}}))
+			g.Expect(tt.expectedResponse).To(EqualObject(response, IgnorePaths{"items", ".message"}))
```
Comment (sbueringer) [inline test/extension/handlers/topologymutation/handler_test.go:307]: Q: Do you know why items doesn't need a leading dot?

Comment (killianmuldoon) [reply test/extension/handlers/topologymutation/handler_test.go:307]: AFAIK it's the embedded types which require the `.` prefix - message is part of CommonResponse so it needs a dot, while items is a top level field by itself.

**Comment by fabriziopandini** [comment]
Nice cleanup
/lgtm

**Pull request merged**